lib/SILGen/SILGenLValue.cpp

```diff
@@ -1344,6 +1351,86 @@ namespace {
       assert(getAccessorDecl()->isSetter());
       SILDeclRef setter = Accessor;
 
+      if (hasPropertyDelegate() && IsOnSelfParameter) {
+        // This is delegated property. Instead of emitting a setter, emit an
+        // assign_by_delegate with the allocating initializer funcction and the
+        // setter function as arguments. DefiniteInitializtion will then decide
+        // between the two functions, depending if it's an initialization or a
+        // re-assignment.
+        //
+        VarDecl *field = dyn_cast<VarDecl>(Storage);
+        VarDecl *backingVar = field->getPropertyDelegateBackingVar();
+        assert(backingVar);
+        CanType ValType = backingVar->getType()->getCanonicalType();
+        SILType varStorageType =
+          SGF.SGM.Types.getSubstitutedStorageType(backingVar, ValType);
+        auto typeData =
+          getLogicalStorageTypeData(SGF.SGM, getTypeData().AccessKind, ValType);
+
+        // Get the address of the storage property.
+        ManagedValue proj;
+        if (BaseFormalType->mayHaveSuperclass()) {
+          RefElementComponent REC(backingVar, LValueOptions(), varStorageType,
+                                  typeData);
+          proj = std::move(REC).project(SGF, loc, base);
+        } else {
+          assert(BaseFormalType->getStructOrBoundGenericStruct());
+          StructElementComponent SEC(backingVar, varStorageType, typeData);
+          proj = std::move(SEC).project(SGF, loc, base);
+        }
+
+        // Create the allocating initializer function. It captures the metadata.
+        auto delegateInfo = getAttachedPropertyDelegateInfo(field);
+        auto ctor = delegateInfo.initialValueInit;
+        SubstitutionMap subs = backingVar->getType()->getMemberSubstitutionMap(
```

Comment:
I think this should be passed in or stored inside the SILGenLValue and not derived from scratch?

Reply:
What do you mean?

Reply:
In the AST when you reference a member of a generic type, Sema records a substitution map in the MemberRefExpr. I think it’s cleaner to use that if you can. Sorry for being vague here, I should read some of the surrounding code in more detail and github’s review interface isn’t so useful for looking at code that’s not part of the PR itself!